irr/src/CGLTFMeshFileLoader.cpp

```diff
@@ -0,0 +1,610 @@
+ * @return vec3 core::Vector3df
+*/
+core::vector3df CGLTFMeshFileLoader::MeshExtractor::readVec3DF(
+		const BufferOffset& readFrom,
+		const core::vector3df scale = {1.0f,1.0f,1.0f})
+{
+	return core::vector3df(
+		readPrimitive<float>(readFrom),
+		readPrimitive<float>(BufferOffset(readFrom, sizeof(float))),
+		-readPrimitive<float>(BufferOffset(readFrom, 2 *
+		sizeof(float))));
+}
+
+/**
+ * Streams vertex positions raw data into usable buffer via reference.
+ * Buffer: ref Vector<video::S3DVertex>
+*/
+void CGLTFMeshFileLoader::MeshExtractor::copyPositions(
+		const std::size_t accessorIdx,
+		std::vector<vertex_t>& vertices) const
+{
+
+	const auto& buffer = getBuffer(accessorIdx);
+	const auto count = getElemCount(accessorIdx);
+	const auto byteStride = getByteStride(accessorIdx);
+
+	for (std::size_t i = 0; i < count; i++) {
+		const auto v = readVec3DF(BufferOffset(buffer, byteStride * i));
+		vertices[i].Pos = v;
+	}
+}
+
+/**
+ * Streams normals raw data into usable buffer via reference.
+ * Buffer: ref Vector<video::S3DVertex>
+*/
+void CGLTFMeshFileLoader::MeshExtractor::copyNormals(
+		const std::size_t accessorIdx,
+		std::vector<vertex_t>& vertices) const
+{
+	const auto& buffer = getBuffer(accessorIdx);
+	const auto count = getElemCount(accessorIdx);
+	
+	for (std::size_t i = 0; i < count; i++) {
+		const auto n = readVec3DF(BufferOffset(buffer,
+			3 * sizeof(float) * i));
+		vertices[i].Normal = n;
+	}
+}
+
+/**
+ * Streams texture coordinate raw data into usable buffer via reference.
+ * Buffer: ref Vector<video::S3DVertex>
+*/
+void CGLTFMeshFileLoader::MeshExtractor::copyTCoords(
+		const std::size_t accessorIdx,
+		std::vector<vertex_t>& vertices) const
+{
+
+	const auto& buffer = getBuffer(accessorIdx);
+	const auto count = getElemCount(accessorIdx);
+
+	for (std::size_t i = 0; i < count; ++i) {
+		const auto t = readVec2DF(BufferOffset(buffer,
+			2 * sizeof(float) * i));
+		vertices[i].TCoords = t;
+	}
+}
+
+/**
+ * The number of elements referenced by this accessor, not to be confused with the number of bytes or number of components.
+ * Documentation: https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_accessor_count
+ * Type: Integer
+ * Required: YES
+*/
+std::size_t CGLTFMeshFileLoader::MeshExtractor::getElemCount(
+		const std::size_t accessorIdx) const
+{
+	return m_gltf_model.accessors->at(accessorIdx).count;
+}
+
+/**
+ * The stride, in bytes, between vertex attributes.
+ * When this is not defined, data is tightly packed.
+ * When two or more accessors use the same buffer view, this field MUST be defined.
+ * Documentation: https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_bufferview_bytestride
+ * Required: NO
+*/
+std::size_t CGLTFMeshFileLoader::MeshExtractor::getByteStride(
+		const std::size_t accessorIdx) const
+{
+	const auto& accessor = m_gltf_model.accessors->at(accessorIdx);
+	// FIXME this does not work with sparse / zero-initialized accessors
```

Comment:
that seems like a pretty big oversight

Reply:
yes, but it's on the wrong branch together with the animated stuff; i'll bring it over

Reply:
https://github.com/minetest/minetest/pull/14557/commits/eea01c3957308c7bc1d0a8fe03dd2324447d778d